src/core/metrics.cc

```diff
@@ -355,13 +362,15 @@ Metrics::InitializeDcgmMetrics()
         LOG_WARNING << "Cannot make field group: " << errorString(dcgmerr);
       }
       dcgmerr = dcgmWatchFields(
-          handle, groupId, fieldGroupId, 2000000 /*update period, usec*/,
+          handle, groupId, fieldGroupId,
+          (long long)(metrics_interval_secs_ * 1e6) /*update period, usec*/,
           5.0 /*maxKeepAge, sec*/, 5 /*maxKeepSamples*/);
       if (dcgmerr != DCGM_ST_OK) {
         LOG_WARNING << "Cannot start watching fields: " << errorString(dcgmerr);
       } else {
         while (!dcgm_thread_exit_.load()) {
-          std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+          std::this_thread::sleep_for(std::chrono::milliseconds(
```

Comment:
You only need to update at 2x the sample rate you use above to avoid missing readings. So this should be 0.5 * the interval.